core/src/main/java/io/substrait/relation/RelProtoConverter.java

```diff
@@ -228,6 +235,37 @@ public Rel visit(ExtensionTable extensionTable) throws RuntimeException {
     return Rel.newBuilder().setRead(builder).build();
   }
 
+  @Override
+  public Rel visit(HashJoin hashJoin) throws RuntimeException {
+    var builder =
+        HashJoinRel.newBuilder()
+            .setCommon(common(hashJoin))
+            .setLeft(toProto(hashJoin.getLeft()))
+            .setRight(toProto(hashJoin.getRight()))
+            .setType(hashJoin.getJoinType().toProto());
+
+    hashJoin
+        .getLeftKeys()
+        .ifPresent(
+            keys -> {
+              for (int i = 0; i < keys.size(); i++) {
+                builder.setLeftKeys(i, toProto(keys.get(i)));
+              }
+            });
+
+    hashJoin
+        .getRightKeys()
+        .ifPresent(
+            keys -> {
+              for (int i = 0; i < keys.size(); i++) {
+                builder.setRightKeys(i, toProto(keys.get(i)));
+              }
+            });
+
```

Comment:
We should also set the post join condition

Reply:
Heh, the Join converter is also missing the post-join condition 😅

Reply:
I will add it too.

Reply:
Updated the code.

Reply:
Thanks for fixing it in both places 🙇